src/databricks/labs/lsql/dashboards.py

```diff
@@ -660,6 +598,177 @@ def _get_query_widget_spec(fields: list[Field], *, frame: WidgetFrameSpec | None
         return spec
 
 
+@dataclass
+class DashboardMetadata:
+    """The metadata defining a lakeview dashboard"""
+
+    display_name: str
+    tile_metadatas: list[TileMetadata] = dataclasses.field(default_factory=list)
+
+    def validate(self) -> None:
+        """Validate the dashboard metadata.
+
+        Raises:
+            ValueError : If the dashboard metadata is invalid.
+        """
+        tile_ids = []
+        for tile in self.tile_metadatas:
+            if not (tile.is_markdown() or tile.is_query()):
+                raise ValueError(f"Tile path is required: {tile}")
+            tile_ids.append(tile.id)
+        counter = collections.Counter(tile_ids)
+        for tile_id, id_count in counter.items():
+            if id_count > 1:
+                raise ValueError(f"Duplicate id: {tile_id}")
+
+    def update(self, other: "DashboardMetadata") -> None:
```

Comment:
```suggestion
    def merge(self, other: "DashboardMetadata") -> None:
```
did you mean merge?

Reply:
Named this after the dict.update, see resources in docstring